src/tools/ceph_dedup_tool.cc

```diff
@@ -551,45 +552,110 @@ class SampleDedupWorkerThread : public Thread
   public:
     using dup_count_t = ssize_t;
 
-    bool find(string& fp) {
+    auto find(string& fp) {
+      auto iter = under_threshold_fp_map.find(fp);
+      if (iter == under_threshold_fp_map.end()) {
+	return over_threshold_fp_map.find(fp);
+      } 
+      return iter;
+    }
+
+    auto is_end() {
+      return over_threshold_fp_map.end();
+    }
```

Comment:
This isn't a bool, it's an iterator.  It's also pretty confusing in use -- I'd like you to rewrite add() now that the maps and fullness check have changed as simply as possible.

Reply:
If you really want to be able to manipulate the under_threshold_fp_map and over_threshold_fp_map as if they were a single structure, move them into a seperate class implementing that data structure with stl-like accessors, iterators, and invariants.  In the future, I expect we'll want to switch it out for something more sophisticated, so having that logic behind an abstraction will be helpful.